### PR TITLE
Add user guide for Flight SQL

### LINKED ISSUE #230: Add user guide section on Flight SQL support
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
We have no docs on Flight SQL support AFAIK

**Describe the solution you'd like**
Add user guide section on Flight SQL support

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by avantgardnerio**
# Which issue does this PR close?

Closes #230.

 # Rationale for this change

Described in the issue, as always.

# What changes are included in this PR?

A user guide.

# Are there any user-facing changes?

They can see the guide.

### CODE REVIEW

File: dev/build-ballista-docker.sh [hunk 1]
```diff
@@ -19,9 +19,7 @@
 
 set -e
 
-docker build -t ballista-builder --build-arg EXT_UID="$(id -u)" -f dev/docker/ballista-builder.Dockerfile .
```
Comment (avantgardnerio) [inline dev/build-ballista-docker.sh:22]: This shouldn't be in here - I must have had git problems... please don't merge until I can review.

**Comment by avantgardnerio** [comment]
When I rebased, some other files got included - this PR should be only adding the guide. Please don't merge until I can resolve.

**Comment by avantgardnerio** [comment]
> some other files got included 

Correction - the build files changed because I separated the rust build from the docker build to make the instructions clearer. I think those changes should go in, and this is indeed ready for merge.

**Review by andygrove — APPROVED** [review]
Thanks @avantgardnerio. I don't have time right now to actually test the instructions but LGTM!

**Pull request merged**